ChefExtensionHandler/bin/chef-install.sh

```diff
@@ -73,12 +89,15 @@ chef_install_from_script(){
       if [ "$chef_version" = "No config file found !!" ]; then
         echo "Configuration error. Azure chef extension Settings file missing."
         exit 1
-      elif [ -z "$chef_version" ]; then
+      elif [ -z "$chef_version" ] && [ -z "$chef_channel" ]; then
```

Comment:
do `chef_channel` and `chef_version` have defaults? Will they ever be empty?

If they do always have defaults of `latest` and `stable`, then this block would never be called.

If they do not always have defaults, then there is a missing block for when we specify channel but not version, and just want to take the latest version available but need to specify a channel like `unstable` to get it.

Reply:
Hi @btm,`chef_channel` and `chef_version` does not have default values.

I have skipped this block for "when we specify channel but not version" as this gets handled in else block "specify channel & version both".

I tested with below code for 
``` elif [ -z "$chef_version" ] && [ ! -z "$chef_channel" ]; then
        echo "Installing chef client from $chef_channel channel"
        sh /tmp/$platform-install.sh -c $chef_channel" 
```
and found that we cannot specify `bootstrap_channel` without `bootstrap_version`. The fact is if we have the value of `chef_channel` as `unstable` or `current` it won`t take `version` default value as `latest` & get the message as `provide a correct version` in log file `/var/log/azure/custom.log`

Hence to avoid getting the same error message in both the block I have to skip this block.

If this is the case then I can avoid it by exiting it with a proper message.

Reply:
> and found that we cannot specify bootstrap_channel without bootstrap_version

Can you expand on this? It works fine for me. 14.0.202 is the latest stable, 14.0.206 is the latest in current right now.

```
root@chef-server:/tmp# sh /tmp/install.sh -c current
ubuntu 16.04 x86_64
Getting information for chef current  for ubuntu...
downloading https://omnitruck.chef.io/current/chef/metadata?v=&p=ubuntu&pv=16.04&m=x86_64
  to file /tmp/install.sh.27773/metadata.txt
trying wget...
sha1    53395a6aa7f9e2d7901e0d775b9f33f947bb14d7
sha256  8a1ccb7f91ee1dfa01fcee8b453c943ea7b4fd8185ab53441cd924fd77770700
url     https://packages.chef.io/files/current/chef/14.0.206/ubuntu/16.04/chef_14.0.206-1_amd64.deb
version 14.0.206
downloaded metadata file looks valid...
downloading https://packages.chef.io/files/current/chef/14.0.206/ubuntu/16.04/chef_14.0.206-1_amd64.deb
  to file /tmp/install.sh.27773/chef_14.0.206-1_amd64.deb
trying wget...
```

Reply:
I was getting download issue for only `-c` option will look into. Thanks @btm

Reply:
Hi @btm, Updated code by adding the missing block for when we specify channel but not version. I guess that was an issue on my local VM which was interrupting to download the latest unstable version. Done testing its working fine now.
Thank you !!